src/dmd/semantic3.d

```diff
@@ -396,7 +396,7 @@ private extern(C++) final class Semantic3Visitor : Visitor
                 if (f.linkage == LINK.d || f.parameterList.length)
                 {
                     // Declare _argptr
-                    Type t = Type.tvalist;
+                    Type t = Type.tvalist.typeSemantic(funcdecl.loc, sc);
```

Comment:
This runs semantic over and over every time tvalist is used. Furthermore, funcdecl.loc is not the location of the declaration of tvalist.

Reply:
> This runs semantic over and over every time tvalist is used.

I'm surprised that `typeSemantic` doesn't cache the result with a `PASS.semanticDone` to the like.

>  Furthermore, funcdecl.loc is not the location of the declaration of tvalist.

It works for LDC and has done so since at least 2016:
https://github.com/ldc-developers/ldc/blob/master/dmd/semantic3.d#L400